mdata/cache/ccache_metric.go

```diff
@@ -202,13 +212,13 @@ func (mc *CCacheMetric) searchBackward(from, until uint32, keys []uint32, res *C
 	}
 
 	for ; ts != 0; ts = mc.chunks[ts].Prev {
+		if ts < from {
+			break
+		}
+
 		log.Debug("CCacheMetric searchBackward: backward search adds chunk ts %d to end", ts)
 		res.End = append(res.End, mc.chunks[ts].Itgen)
 		res.Until = ts
-
-		if ts <= from {
-			break
-		}
```

Comment:
That's mentioned in the last sentence of the commit message. Moving that up fixed a bug which made `searchBackward()` seek one chunk too far: https://github.com/raintank/metrictank/pull/455/commits/ecf02ee32354c6cd90e7031acc0dcc88071dd0b4